niceplots/utils.py

```diff
@@ -197,6 +197,59 @@ def draggable_legend(axis=None, color_on=True, **kwargs):
         legend[idx].draggable()
 
 
+def label_line_ends(ax, lines=None, labels=None, colors=None, **kwargs):
+    """Place a label just to the right of each line in the axes
+
+    Note: Because the labels are placed outside of the axes, this function works best for plots where all lines end as
+    close to the right edge of the axes as possible. Additionally you need to either use constrained_layout=True
+    (as NicePlots styles do), or call plt.tight_layout() after calling this function.
+
+    Parameters
+    ----------
+    ax : Matplotlib axes
+        axes to label the lines of
+    lines : iterable of matplotlib line objects, optional
+        Lines to label, by default all lines in the axes
+    labels : list of strings, optional
+        Labels for each line, by default uses each line's label attribute
+    colors : single or list of colors, optional
+        Color(s) to use for each line, can be a single color for all lines or a list containing an entry for each line,
+        by default uses each line's color
+    **kwargs :
+        Any valid keywords for matplotlib's annotate function
```

Comment:
Would be nice to document the keywords that can't be included since they would generate errors (`xy`, `xytext`, `color`, etc.)

Reply:
Done